process/sync/baseSync.go

```diff
@@ -82,9 +85,19 @@ func (boot *baseBootstrap) loadBlocks(
 
 	log.Info(fmt.Sprintf("the highest header nonce committed in storer is %d\n", highestNonceInStorer))
 
+	var finalNotarized map[uint32]uint64
+	var lastNotarized map[uint32]uint64
+
+	shardId := boot.shardCoordinator.SelfId()
+
 	for currentNonce = highestNonceInStorer; currentNonce > blockFinality; currentNonce-- {
+		currentNonce, finalNotarized, lastNotarized = getNonceWithLastNotarized(currentNonce)
```

Comment:
This should be refactored

Reply:
Done

Reply:
Not done, you update a for counter in the for body